core-spi-test/src/main/java/org/ehcache/internal/store/StorePutIfAbsentTest.java

```diff
@@ -165,9 +165,8 @@ public void wrongValueTypeThrowsException()
         kvStore2.putIfAbsent(key, "value");
       }
       throw new AssertionError("Expected ClassCastException because the value is of the wrong type");
-    } catch (ClassCastException e) {
-      // expected
-    } catch (CacheAccessException e) {
+    } catch (CacheAccessException cae) {
+      assertThat(cae.getCause(), instanceOf(ClassCastException.class));
```

Comment:
Wait, that changes the test semantics. You should do nothing if the cause is a `ClassCastException` and throw otherwise.